tests/unit/data/test_datapoint.py

```diff
@@ -58,15 +34,14 @@ def features_generators():
 
 
 def test_features_and_fg(features_generators):
-    smi = "c1ccccc1"
     targets = np.random.rand(1, 1)
     features = np.random.rand(1024)
     with pytest.raises(ValueError):
-        MoleculeDatapoint.from_smi(smi, y=targets, x_f=features, mfs=features_generators)
+        MoleculeDatapoint.from_smi(SMI, y=targets, x_f=features, mfs=features_generators)
 
 
 def test_num_tasks(targets):
-    d = MoleculeDatapoint.from_smi("c1ccccc1", y=targets)
+    d = MoleculeDatapoint.from_smi(SMI, y=targets)
 
     assert d.t == targets.shape[0]
```

Comment:
`targets` is set such that this test runs with 1 task and 2 tasks, which is wanted. The next test, `test_addh` is run twice for each SMILES because `target` is still set to have two values. I think `targets` could be better defined in `test_addh` and `test_replace_token` as is done in `test_features_and_fg` so they are run half as many times. Perhaps that is outside the scope of your PR.

Reply:
I think it's out of scope. I'm not trying to rewrite test logic, just abstract out fixtures where possible